src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,1591 @@
+            c.M(i3);
+            ((I)c).M(i3);
+            """;
+
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            public interface I1 {}
+            public interface I2 {}
+            public interface I3 : I1, I2 {}
+
+            public interface I
+            {
+                [OverloadResolutionPriority(1)]
+                void M(I1 i1);
+                void M(I2 i2);
+            }
+            public class C : I
+            {
+                public void M(I1 i1) => System.Console.Write(1);
+                [OverloadResolutionPriority(2)]
+                public void M(I2 i2) => System.Console.Write(2);
+            }
+            """;
+
+        CompileAndVerify([executable, source, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "21").VerifyDiagnostics();
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CompileAndVerify(executable, references: [useMetadataReference ? comp.ToMetadataReference() : comp.EmitToImageReference()], expectedOutput: "21").VerifyDiagnostics();
+    }
+
+    [Theory, CombinatorialData]
+    public void Interface_DifferentPriorities_Indexers(bool useMetadataReference)
+    {
+        var executable = """
+            var c = new C();
+            I3 i3 = null;
+            c[i3] = 1;
+            _ = c[i3];
+            ((I)c)[i3] = 1;
+            _ = ((I)c)[i3];
+            """;
+
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            public interface I1 {}
+            public interface I2 {}
+            public interface I3 : I1, I2 {}
+
+            public interface I
+            {
+                [OverloadResolutionPriority(1)]
+                int this[I1 i1] { get; set; }
+                int this[I2 i2] { get; set; }
+            }
+            public class C : I
+            {
+                public int this[I1 i1]
+                {
+                    get
+                    {
+                        System.Console.Write(1);
+                        return 1;
+                    }
+                    set => System.Console.Write(2);
+                }
+                [OverloadResolutionPriority(1)]
+                public int this[I2 i2]
+                {
+                    get
+                    {
+                        System.Console.Write(3);
+                        return 1;
+                    }
+                    set => System.Console.Write(4);
+                }
+            }
+            """;
+
+        CompileAndVerify([executable, source, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "4321").VerifyDiagnostics();
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CompileAndVerify(executable, references: [useMetadataReference ? comp.ToMetadataReference() : comp.EmitToImageReference()], expectedOutput: "4321").VerifyDiagnostics();
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Let's test explicit interface implementations, where the attribute is either the interface method, or the explicit implementation, or both.